sdk/core/core/inc/az_http_response_parser.h

```diff
@@ -61,14 +49,28 @@ typedef struct {
  * Initializes an HTTP response parser.
  */
 AZ_NODISCARD az_result
-az_http_response_parser_init(az_span const buffer, az_http_response_parser * const out);
+az_http_response_parser_init(az_http_response_parser * const out, az_span const buffer);
+
+/**
+ * An HTTP status line.
+ */
+AZ_NODISCARD az_result az_http_response_parser_get_status_line(
+    az_http_response_parser * const self,
+    az_http_response_status_line * const out);
+
+/**
+ * An HTTP header.
+ */
+AZ_NODISCARD az_result az_http_response_parser_get_header(
+    az_http_response_parser * const self,
+    az_http_response_header * const out);
```

Comment:
Shouldn't this just be an az_pair?
Do we need a special az_http_response_header data type?